src/store/preferences.ts

```diff
@@ -112,6 +124,32 @@ export const actions = {
   },
   setPlatformWindows({ commit }: PrefActionContext, isPlatformWindows: boolean) {
     commit('SET_IS_PLATFORM_WINDOWS', isPlatformWindows);
+  },
+  async proposePreferences({ commit, state }: PrefActionContext, { port, user, password }: {port: number, user: string, password: string}) {
+    const result = await fetch(
+      proposedSettings(port),
+      {
+        method:  'PUT',
+        headers: new Headers({
+          Authorization:  `Basic ${ window.btoa(`${ user }:${ password }`) }`,
+          'Content-Type': 'application/x-www-form-urlencoded'
+        }),
+        body: JSON.stringify(state.preferences)
+      });
+
+    const severities = Object.entries(await result.json())
+      .map(([_key, val]: [key: string, val: any]) => {
+        const { severity } = val;
+
+        return { severity };
+      })
+      .reduce((prev, curr) => {
+        return { ...prev, [curr.severity]: true };
+      }, { reset: false, restart: false });
```

Comment:
Testing out alternatives…
```suggestion
    const changes: Record<string, {severity: 'reset' | 'restart'}> = await result.json();
    const values = Object.values(changes).map(v => v.severity);
    const severities = {
      reset:   values.includes('reset'),
      restart: values.includes('restart'),
    };
```
I'm not actually sure the alternative is better; maybe a combination of the two?

Reply:
I think your recommendation is quite elegant and easy to read, thanks 🙂